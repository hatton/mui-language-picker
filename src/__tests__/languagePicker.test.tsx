import React from 'react';
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { LanguagePicker, languagePickerStrings_en } from '..';

afterEach(cleanup);

const MyComponent: React.FC = (props: any) => {
  const [bcp47, setBcp47] = React.useState('und');
  const [lgName, setLgName] = React.useState('');
  const [fontName, setFontName] = React.useState('');

  return (
    <LanguagePicker
      value={bcp47}
      setCode={setBcp47}
      name={lgName}
      setName={setLgName}
      font={fontName}
      setFont={setFontName}
      t={languagePickerStrings_en}
    />
  );
};

test('can render Language Picker snapshot', async () => {
  const { getByText, container } = render(<MyComponent />);
  await waitForElement(() => getByText(/^Language$/i));
  expect(container).toMatchSnapshot();
});
