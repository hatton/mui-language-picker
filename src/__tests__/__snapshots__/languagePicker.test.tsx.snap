// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render Language Picker snapshot 1`] = `
<div>
  <div
    id="LangBcp47"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense"
      style="width: 300px;"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-filled Mui-required Mui-required"
        data-shrink="false"
        for="lang-bcp47"
        id="lang-bcp47-label"
      >
        Language
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
          â€‰
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
          id="lang-bcp47"
          required=""
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
